File: Programmers/C++/올바른 괄호.cpp
#include <string>
#include <iostream>

using namespace std;

int n;

bool solution(string s)
{
    for(auto c: s) {
        if(c=='(') ++n;
        else {
            if(--n<0) return false;
        }
    }

    return n==0? true : false;
}

File: Programmers/Swift/올바른 괄호.swift
import Foundation

func solution(_ s:String) -> Bool
{
    var n = 0
    for i in s {
        if i=="(" {n+=1}
        else {
            n-=1
            if n<0 {return false}
        }
    }
    if n==0 {return true}
    else {return false}
}
